torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -1457,6 +1457,7 @@ void ProcessGroupNCCL::heartbeatMonitor() {
     // Leave another two mins for desync report generation or process group
     // destroy.
     std::this_thread::sleep_for(std::chrono::seconds(heartbeatTimeoutInSec_));
+    LOG(INFO) << logPrefix() << "slept for " << heartbeatTimeoutInSec_;
```

Comment:
should we be logging why we're sleeping?

Reply:
Good point! Added more clarification!